### PR TITLE
Fix bugs when running e2e test on the pre-exiting kubemark cluster

### PR DESCRIPTION
**Pull request by Lily922**
**What this PR does / why we need it**:

1. When I running e2e tests on the kubemark cluster in the "pre-exiting" way, I got an error like "run-e2e-tests.sh: line 29: detet-master: command not found" during running the script named "kubernetes/test/kubemark/run-e2e-test.sh". I fixed it by adding "detect-master" function to kubernetes/cluster/pre-existing/util.sh
2. I also go an error like "run-e2e-tests.sh: line 41: ENABLE_KUBEMARK_CLUSTER_AUTOSCALER:  unbound variable". I fixed it by adding variable "ENABLE_KUBEMARK_CLUSTER_AUTOSCALE" to kubemark/cluster/kubemark/pre-existing/config-default.sh
   I am not sure if these solutions are completely corrent. I think these bugs are because someone added these variables to the scripts which used to run tests on kubemark cluster building by gce, but forgot to update the scripts used for the "pre-exiting" way at the same time . All of my operation is on the branch 1.8.3. 
   Another PR I have submitted is [#56193.](https://github.com/kubernetes/kubernetes/pull/56193)
   Thanks and sorry for my poor English. 
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: cluster/pre-existing/util.sh [hunk 1]
```diff
@@ -60,3 +60,18 @@ function create-certs {
   KUBEAPISERVER_CERT_BASE64=$(dd if=/dev/urandom bs=128 count=1 2>/dev/null | base64 | tr -d "=+/" | dd bs=32 count=1 2>/dev/null)
   KUBEAPISERVER_KEY_BASE64=$(dd if=/dev/urandom bs=128 count=1 2>/dev/null | base64 | tr -d "=+/" | dd bs=32 count=1 2>/dev/null)
 }
+function detect-master() {
+  detect-project
+  KUBE_MASTER=${MASTER_NAME}
+  echo "Trying to find master named '${MASTER_NAME}'" >&2
+  if [[ -z "${KUBE_MASTER_IP-}" ]]; then
```
Comment (roberthbailey) [inline cluster/pre-existing/util.sh]: nit: should be `"${KUBE_MASTER_IP:-}"`

File: cluster/pre-existing/util.sh [hunk 2]
```diff
@@ -60,3 +60,18 @@ function create-certs {
...
+    local master_address_name="${MASTER_NAME}-ip"
+    echo "Looking for address '${master_address_name}'" >&2
+    KUBE_MASTER_IP=${MASTER_IP}
```
Comment (roberthbailey) [inline cluster/pre-existing/util.sh]: where does `$MASTER_IP` get set? The line before echos that you are looking for `$master_address_name` but I don't see anything that actually looks for it.

File: cluster/pre-existing/util.sh [hunk 3]
```diff
@@ -60,3 +60,18 @@ function create-certs {
...
+  fi
+  if [[ -z "${KUBE_MASTER_IP-}" ]]; then
```
Comment (roberthbailey) [inline cluster/pre-existing/util.sh]: nit: should be `"${KUBE_MASTER_IP:-}"`

Comment (LilyFaFa) [reply cluster/pre-existing/util.sh]: You can refer to the following link https://github.com/kubernetes/kubernetes/tree/master/test/kubemark/pre-existing
It introduced how to change the related configuration file and it provided an example at the end.

File: cluster/pre-existing/util.sh (hunk 2 above)
Comment (LilyFaFa) [reply cluster/pre-existing/util.sh]: You can refer to the following link https://github.com/kubernetes/kubernetes/tree/master/test/kubemark/pre-existing
It introduced how to change the related configuration file and it provided an example at the end.

Comment (roberthbailey) [reply cluster/pre-existing/util.sh]: /cc @gmarek - can you verify that this looks correct?

**Comment by roberthbailey** [comment]
ping @gmarek for review.

**Comment by roberthbailey** [comment]
@LilyFaFa - I updated your initial comment to properly specify that this change doesn't require a release note (which would otherwise block it from merging).

File: cluster/pre-existing/util.sh (hunk 2 above)
Comment (gmarek) [reply cluster/pre-existing/util.sh]: @shyamjvs is current owner

**Comment by shyamjvs** [comment]
I'll take a look in a while.

/assign

File: cluster/kubemark/pre-existing/config-default.sh [hunk 4]
```diff
@@ -37,3 +37,4 @@ KUBELET_TEST_LOG_LEVEL="${KUBELET_TEST_LOG_LEVEL:-}"
 KUBEPROXY_TEST_LOG_LEVEL="${KUBEPROXY_TEST_LOG_LEVEL:-}"
 MASTER_NAME="${MASTER_NAME:-}"
 USE_REAL_PROXIER="${USE_REAL_PROXIER:-true}"
+ENABLE_KUBEMARK_CLUSTER_AUTOSCALER="${ENABLE_KUBEMARK_CLUSTER_AUTOSCALER:-false}"
```
Comment (shyamjvs) [inline cluster/kubemark/pre-existing/config-default.sh]: I don't think this variable should be added here (as your master is anyway assumed to be pre-existing). Instead, could you please change [this line](https://github.com/kubernetes/kubernetes/blob/master/test/kubemark/run-e2e-tests.sh#L41) in run-e2e-tests.sh to not fail if the variable is not defined. That is,

```
"${ENABLE_KUBEMARK_CLUSTER_AUTOSCALER:-}"
```

**Comment by Kevin-fqh** [comment]
@LilyFaFa could you describe some details about how to build a kubemark cluster in pre-existing way?

**Comment by shyamjvs** [comment]
Do we still need this PR?

**Comment by shyamjvs** [comment]
Closing this as it seems stale. Re-open if needed.

**Pull request closed without merging**